macaroon.js

```diff
@@ -338,197 +392,249 @@ const readFieldV2Optional = function(buf, maybeFieldType) {
-  @param {bitArray} ciphertext An sjcl bitArray as returned by encrypt.
+  Decrypts the given cyphertext.
+  @param {bitArray} keyBits decryption key.
+  @param {bitArray} ciphertextBits encrypted text.
+  @return {bitArray} decrypted text.
 */
-function decrypt(key, ciphertext) {
-  key = bitArrayToUint8Array(key);
-  ciphertext = bitArrayToUint8Array(ciphertext);
-  const nonce = ciphertext.slice(0, NONCELEN);
-  ciphertext = ciphertext.slice(NONCELEN);
-  var text = nacl.secretbox.open(ciphertext, nonce, key);
-  if (text === false) {
+const decrypt = function(keyBits, ciphertextBits) {
+  const keyBytes = bitsToBytes(keyBits);
+  const ciphertextBytes = bitsToBytes(ciphertextBits);
+  const nonceBytes = ciphertextBytes.slice(0, NONCELEN);
+  const dataBytes = ciphertextBytes.slice(NONCELEN);
+  var textBytes = nacl.secretbox.open(dataBytes, nonceBytes, keyBytes);
+  if (textBytes === false) {
     throw new Error('decryption failed');
   }
-  return bytesToBitArray(text);
-}
+  return bytesToBits(textBytes);
+};
 
-const zeroKey = sjcl.codec.hex.toBits('0'.repeat(64));
+const zeroKeyBits = bytesToBits(stringToBytes('\0'.repeat(32)));
 
 /**
   Bind a given macaroon to the given signature of its parent macaroon. If the
   keys already match then it will return the rootSig.
-  @param {bitArray} rootSig
-  @param {bitArray} dischargeSig
-  @return {bitArray} The bound macaroon.
+  @param {bitArray} rootSigBits
+  @param {bitArray} dischargeSigBits
+  @return {bitArray} The bound macaroon signature.
 */
-function bindForRequest(rootSig, dischargeSig) {
-  if (sjcl.bitArray.equal(rootSig, dischargeSig)) {
-    return rootSig;
+const bindForRequest = function(rootSigBits, dischargeSigBits) {
+  if (sjcl.bitArray.equal(rootSigBits, dischargeSigBits)) {
+    return rootSigBits;
   }
-  return keyedHash2(zeroKey, rootSig, dischargeSig);
-}
+  return keyedHash2(zeroKeyBits, rootSigBits, dischargeSigBits);
+};
 
 const Macaroon = class Macaroon {
   /**
     Create a new Macaroon with the given root key, identifier, location
     and signature.
     @param {Object} The necessary values to generate a macaroon.
       It contains the following fields:
-        identifier: {String}
-        location:   {String}
-        caveats:    {Array}
-        signature:  {bitarray}
+        identifierBytes: {Uint8Array}
+        locationStr:   {string}
+        caveats:    {Array of {locationStr: string, identifierBytes: Uint8Array, vidBytes: Uint8Array}}
+        signatureBytes:  {Uint8Array}
+        version: {int} The version of macaroon to create.
   */
-  constructor({identifier, location, caveats, signature}) {
-    this._location = location;
-    this._identifier = identifier;
-    this._signature = signature;
-    this._caveats = caveats;
+  constructor(params) {
+    if (!params) {
+      // clone uses null parameters.
+      return;
+    }
+    var {version, identifierBytes, locationStr, caveats, signatureBytes} = params;
```

Comment:
const or let

Reply:
done (I didn't know about let when I wrote that :-])